### ISSUE TITLE
DPMSolverSinglestepScheduler with use_karras_sigmas=True triggers RuntimeError: a Tensor with 2 elements cannot be converted to Scalar 

### ISSUE THREAD

**Issue by alexisrolland** [huggingface/diffusers#4631]
### Describe the bug

Hello @patrickvonplaten and @williamberman 🤗

Running `StableDiffusionPipeline` with `DPMSolverSinglestepScheduler` and `use_karras_sigmas=True` triggers the error `RuntimeError: a Tensor with 2 elements cannot be converted to Scalar ` when generating images with a high `num_inference_steps` (~100). Note this issue does not happen with low `num_inference_steps` (~20).

### Reproduction

```py
import os
import torch
from diffusers import StableDiffusionPipeline, DPMSolverSinglestepScheduler

MODEL_PATH = os.getenv('MODEL_PATH')
pipeline = StableDiffusionPipeline.from_pretrained(MODEL_PATH, torch_dtype=torch.float16, variant='fp16')
pipeline.scheduler = DPMSolverSinglestepScheduler.from_pretrained(MODEL_PATH, subfolder='scheduler', use_karras_sigmas=True)
pipeline.enable_model_cpu_offload()

prompts = ["a cat playing with a ball in the forest"]
negative_prompts =["bad quality"]
seed = 123456
generators = torch.Generator(device='cuda').manual_seed(seed)

image = pipeline(
    prompt=prompts,
    negative_prompt=negative_prompts,
    width=800,
    height=1200,
    num_inference_steps=100,
    guidance_scale=6,
    num_images_per_prompt=1,
    generator=generators
).images[0].save("image.png")
```

### Logs

```shell
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py", line 688, in __call__
    latents = self.scheduler.step(noise_pred, t, latents, **extra_step_kwargs, return_dict=False)[0]
  File "/usr/local/lib/python3.10/site-packages/diffusers/schedulers/scheduling_dpmsolver_singlestep.py", line 670, in step
    step_index = step_index.item()
RuntimeError: a Tensor with 2 elements cannot be converted to Scalar
```

### System Info

- `diffusers` version: 0.19.3
- Platform: Linux-5.4.72-microsoft-standard-WSL2-x86_64-with-glibc2.36
- Python version: 3.10.12
- PyTorch version (GPU?): 2.0.1+cu117 (True)
- Huggingface_hub version: 0.16.4
- Transformers version: 4.30.2
- Accelerate version: 0.21.0
- xFormers version: 0.0.20
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No

**Comment by sayakpaul** [comment]
Cc: @yiyixuxu too.

**Comment by yiyixuxu** [comment]
thanks for the issue. It is caused by duplicated timesteps and will be fixed by https://github.com/huggingface/diffusers/pull/4187
(I will start work on this today!)

**Comment by CodeCorrupt** [comment]
Hey, I wanted to bump this issue since it seems the linked PR is for `DPMSolverMultistepScheduler` and I'm currently seeing the issue with `DPMSolverSinglestepScheduler` when `use_karras_sigmas=True`. 
Also of note is that I'm only seeing the error when the step count is > ~10 steps.

**Comment by patrickvonplaten** [comment]
Linked PR: https://github.com/huggingface/diffusers/pull/4986

**Comment by yiyixuxu** [comment]
fixed with https://github.com/huggingface/diffusers/pull/4986

**Comment by Z-YuPeng** [comment]
karras_dpmpp_sde still has this bug in the latest diffuser version.

**Comment by patrickvonplaten** [comment]
cc @yiyixuxu could you check here?

**Comment by alexisrolland** [comment]
I confirm this is fixed as I've tested this again on 0.22.2.

**Comment by burgalon** [comment]
Here's some code to reproduce
```import torch
from diffusers import StableDiffusionPipeline, StableDiffusionControlNetInpaintPipeline, DPMSolverMultistepScheduler, ControlNetModel
from diffusers.utils import load_image
control = ControlNetModel.from_pretrained(torch_dtype=torch.float16, pretrained_model_name_or_path="lllyasviel/control_v11p_sd15_openpose").to("cuda")
base = StableDiffusionPipeline.from_pretrained("/data/models/690204-sd15", torch_dtype=torch.float16).to("cuda")
pipe = StableDiffusionControlNetInpaintPipeline(
    vae=base.vae,
    text_encoder=base.text_encoder,
    tokenizer=base.tokenizer,
    unet=base.unet,
    scheduler=base.scheduler,
    safety_checker=None,
    feature_extractor=base.feature_extractor,
    controlnet=control,
).to('cuda')
scheduler = DPMSolverMultistepScheduler.from_config(pipe.scheduler.config, algorithm_type="sde-dpmsolver++", use_karras_sigmas=True)
pipe.scheduler = scheduler
image = load_image('https://sdbooth2-production.s3.amazonaws.com/pph45nwtz5w0afa8nlhbt41djwep')
pipe(prompt="test", image=image, mask_image=image, control_image=image)
```


exception
```
File /opt/venv/lib/python3.8/site-packages/diffusers/schedulers/scheduling_dpmsolver_multistep.py:888, in DPMSolverMultistepScheduler.add_noise(self, original_samples, noise, timesteps)
    885     schedule_timesteps = self.timesteps.to(original_samples.device)
    886     timesteps = timesteps.to(original_samples.device)
--> 888 step_indices = [(schedule_timesteps == t).nonzero().item() for t in timesteps]
    890 sigma = sigmas[step_indices].flatten()
    891 while len(sigma.shape) < len(original_samples.shape):

File /opt/venv/lib/python3.8/site-packages/diffusers/schedulers/scheduling_dpmsolver_multistep.py:888, in <listcomp>(.0)
    885     schedule_timesteps = self.timesteps.to(original_samples.device)
    886     timesteps = timesteps.to(original_samples.device)
--> 888 step_indices = [(schedule_timesteps == t).nonzero().item() for t in timesteps]
    890 sigma = sigmas[step_indices].flatten()
    891 while len(sigma.shape) < len(original_samples.shape):

RuntimeError: a Tensor with 2 elements cannot be converted to Scalar```
